xcdat/averager.py

```diff
@@ -20,275 +23,311 @@
     regional selection.
 
     Operations include:
 
-    - Generate weights for axes using axis bounds
+    - Generate weights for axes using dimension bounds.
+       - If dimension doesn't have bounds, weights aren't calculated for it.
     - If a region is specified, calculate the weights corresponding to the
-    regional bounds
+       regional bounds.
+    - Latitude weights are scaled with sine(latitude).
 
     Parameters
     ----------
     xvariable : xr.DataArray
         xr.DataArray produced using ``open_variable``
-    lat_bounds : list, optional
-        list of latitude boundaries for regional selection (e.g.,
-        lat_bounds=[-5, 5])
-    lon_bounds : list, optional
-        list of longitude boundaries for regional selection (e.g.,
-        lon_bounds=[-170, -120])
-    weight : str, optional
-        Axis weighting option (either "weighted" or "unweighted"), default
-        "weighted"
+    lat_bounds : Optional[RegionAxisBounds], optional
+        List of latitude boundaries for regional selection (e.g.,
+        lat_bounds=[-5, 5]), by default None
+    lon_bounds : Optional[RegionAxisBounds], optional
+        List of longitude boundaries for regional selection (e.g.,
+        lon_bounds=[-170, -120]), by default None
+    is_weighted : bool, True
+        Weighting option to calculate weights proportional to area
+        ("weighted"), uniform weights ("unweighted").
 
     Returns
     -------
-    dict
+    AxisWeights
         Set of weights for each eligible dimension.
 
     Notes
     -----
     This is a prototype and needs work to make it work more generically. It
     currently only works with lat / lon. It might be useful to check the
     latitude axis units.
+    """
+    if is_weighted:
+        weights = _get_area_weights(xvariable, lat_bounds, lon_bounds)
+    elif not is_weighted:
+        weights = _get_uniform_weights(xvariable)
+    return weights
 
-    Examples
-    --------
-    Import and call module:
-    >>> from xcdat.dataset import open_dataset
-    >>> from xcdat.variable import open_variable
-    >>>
-    >>> filename = 'test.nc'
-    >>> ds = open_dataset(filename)
-    >>> tas = open_variable(ds, 'tas')
-    >>> axis_weights = _get_axis_weights(tas)
 
-    Raises
-    ------
-    ValueError
-        Incorrect weight option specified
+def _get_area_weights(
+    xvariable: xr.DataArray,
+    lat_bounds: Optional[RegionAxisBounds],
+    lon_bounds: Optional[RegionAxisBounds],
+):
+    dims_bounds: Dict[str, Optional[xr.DataArray]] = xvariable.bounds.bounds
+    # TODO: Make this generic, currently only able to handle lat/lon
+    regions_bounds: Dict[str, Optional[List[float]]] = {
+        "lat": lat_bounds,
+        "latitude": lat_bounds,
+        "lon": lon_bounds,
+        "longitude": lon_bounds,
+    }
+    weights: AxisWeights = {}
+
+    for dim_name, bounds in dims_bounds.items():
+        if bounds is not None and dim_name in SUPPORTED_AXES:
```

Comment:
TODO: swap conditionals around so they it is more readable and can exit more quickly if needed